include/nbl/video/IDeviceMemoryBacked.h

```diff
@@ -84,7 +117,12 @@ class IDeviceMemoryBacked : public virtual core::IReferenceCounted
         {
             const uint32_t* queueFamilyIndices = nullptr;
         };
-
+        
+        void chainPreDestroyCleanup(std::unique_ptr<ICleanup> first)
+        {
+            ICleanup::chain(m_cachedCreationParams.preDestroyCleanup, std::move(first));
+        }
```

Comment:
the cleanups are not meant to be dynamic, all the cleanups you'll ever need are required to be specified in the SCreationParams

Reply:
the fact you've added this method tells me one thing, you're attempting to use the system to make the Nabla object hold onto something _after it has already been created_, probably the CUDA resource that you've created from it. Esp because you're hooking into the pre-destroy cleanup.

Whereas the intended use is for Nabla object to hold onto some external resoruce it has been created from.

In the case where you have a Nabla object, owned and created by Nabla, being exported, which created an object in some other API that needs to be destroyed before the Nabla object...

The correct way is for the other API object wrapper to hold a `smart_Refctd_ptr` to the Nabla object it was imported from.